tcms/testruns/static/testruns/js/get.js

```diff
@@ -2,4 +2,14 @@ $(document).ready(() => {
 
     $('.bootstrap-switch').bootstrapSwitch();
 
+    const testRunId = $('#test_run_pk').data('pk')
+
+    $('#status_button').on('switchChange.bootstrapSwitch', (_event, state) => {
+        if (state) {
+            jsonRPC('TestRun.update', [testRunId, { 'stop_date': null }], () => { })
```

Comment:
This doesn't work because of  these few lines in TestRun.update API method:

```
    if form.cleaned_data['stop_date']:
        test_run.stop_date = form.cleaned_data['stop_date']
```

The entire method is badly written b/c it declares and handles form fields one by one. Instead the underlying form used for data validation should inherit from ModelForm (see #1384) and become very simple, similar to TestCase.update method.  This is an independent change but needs to happen before this button can be implemented.